CMakeLists.txt

```
@@ -130,23 +130,15 @@ if(UR_USE_MSAN)
     add_sanitizer_flag(memory)
 endif()
 
-# Allow custom third_party folder
-if(NOT DEFINED THIRD_PARTY_DIR)
-    set(THIRD_PARTY_DIR ${CMAKE_CURRENT_SOURCE_DIR}/third_party)
-endif()
-
-find_program(CLANG_FORMAT NAMES clang-format-15 clang-format-15.0 clang-format)
-
-set(CLANG_FORMAT_REQUIRED "15.0")
+# Check if clang-format (in correct version) is available for Cpp code formatting.
+if(UR_FORMAT_CPP_STYLE)
+    find_program(CLANG_FORMAT NAMES clang-format-15 clang-format-15.0 clang-format)
```

I think it would make sense to use [CMAKE_EXECUTABLE_SUFFIX](https://cmake.org/cmake/help/latest/variable/CMAKE_EXECUTABLE_SUFFIX.html) here.

```suggestion
    find_program(CLANG_FORMAT NAMES
        clang-format-15${CMAKE_EXECUTABLE_SUFFIX}
        clang-format-15.0${CMAKE_EXECUTABLE_SUFFIX}
        clang-format${CMAKE_EXECUTABLE_SUFFIX})
```

---

It seems that `find_program` is handling this on its own. I couldn't confirm it in official CMake docs, but here's my finding: https://www.scivision.dev/cmake-find-program-script/

And, of course, our CI is working as expected: https://github.com/oneapi-src/unified-runtime/actions/runs/5891285675/job/15978032388#step:5:35

---

ok, I see, thanks - I'll keep that in mind, in case it pops up somewhere